include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,690 @@
+        //!\cond
+            requires std::RandomAccessIterator<underlying_iterator_type>
+        //!\endcond
+        {
+            size_t src_size = end_it - begin_it;
+            size_t index_i = src_size - 2 -
+                             std::floor(std::sqrt(-8 * index + 4 * src_size * (src_size - 1) - 7)/2.0 - 0.5);
+            size_t index_j = index + index_i + 1 - src_size * (src_size - 1)/2 + (src_size - index_i) *
+                             ((src_size - index_i) - 1)/2;
+            first_it = begin_it + index_i;
+            second_it = begin_it + index_j;
+        }
+
+        //!\brief The iterator pointing to the first element of the pairwise combination.
+        underlying_iterator_type first_it{};
+        //!\brief The iterator pointing to the second element of the pairwise combination.
+        underlying_iterator_type second_it{};
+        //!\brief The begin of the underlying range.
+        underlying_iterator_type begin_it{};
+        //!\brief The end of the underlying range.
+        underlying_iterator_type end_it{};
+    };
+
+public:
+
+    //!\name Member types
+    //!\{
+    //!\brief The iterator type.
+    using iterator = iterator_type<underlying_range_type>;
+    //!\brief The const iterator type. Evaluates to void if the underlying range is not const iterable.
+    using const_iterator = transformation_trait_or_t<std::type_identity<iterator_type<underlying_range_type const>>,
+                                                     void>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr pairwise_combine_view() = default;
+    constexpr pairwise_combine_view(pairwise_combine_view const &) = default;
+    constexpr pairwise_combine_view(pairwise_combine_view &&) = default;
+    constexpr pairwise_combine_view & operator=(pairwise_combine_view const &) = default;
+    constexpr pairwise_combine_view & operator=(pairwise_combine_view &&) = default;
+    ~pairwise_combine_view() = default;
+
+    /*!\brief Constructs from a view.
+     * \param[in] range The underlying range to be wrapped. Of type `underlying_range_type`.
+     *
+     * \details
+     *
+     * During construction the iterator pointing to the last element of the view is cached (not the end of the range).
+     * This optimises the call to end if the underlying range models only ForwardRange. Otherwise the call to end will be
+     * linear in the number of elements of the underlying range.
+     *
+     * \attention This view cannot be chained immediately after an infinite range, because upon construction it will
+     *            take forever to reach the last element of the view.
+     *
+     * ### Complexity
+     *
+     * Constant if `underlying_range_type` models std::ranges::BidirectionalRange, otherwise linear.
+     */
+    explicit constexpr pairwise_combine_view(underlying_range_type range) : src_range{std::move(range)}
```

Comment:
If you want to keep this explicit, can you make a card for the range module that we need to decide on doing this eerywhere or nowhere?